### PR TITLE
Audit fixes brian

### PR DESCRIPTION
**Pull request by brossetti1**
wip fixing quantstamp issues

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "solidity.defaultCompiler": "localNodeModule",
```
Comment (brossetti1) [inline .vscode/settings.json:2]: oops should proably update this?

**Comment by dekanbro** [comment]
updated docs for some of these issues
QSP-5 Ownership Can Be Renounced
QSP-6 Shamans Can Be an EOA Address
QSP-10 Risk of Killing Upgrades
QSP-12 A DAO's Safety Is Dependent on the Safety of Its Shamans
QSP-13 Upgradability
QSP-14 msg.sender Can Be Overridden.
QSP-15 External Calls to Malicious Contracts

https://github.com/HausDAO/baal-docs/commit/5151a944471c8aa74ed77139ec66b0de697aaf96

File: contracts/Baal.sol [hunk 2]
```diff
@@ -14,16 +14,15 @@ import "@gnosis.pm/safe-contracts/contracts/GnosisSafe.sol";
...
+import "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
 
 import "./interfaces/IBaalToken.sol";
 
 /// @title Baal ';_;'.
 /// @notice Flexible guild contract inspired by Moloch DAO framework.
-contract Baal is Module, EIP712, ReentrancyGuard, BaseRelayRecipient {
-    using ECDSA for bytes32;
+contract Baal is Module, EIP712Upgradeable, ReentrancyGuardUpgradeable, BaseRelayRecipient {
+    using ECDSAUpgradeable for bytes32;
```
Comment (brossetti1) [inline contracts/Baal.sol:25]: so this is actually an upgrade on the ECDSA logic - there was some signature validations that openzeppelin removed from the ECDSA checks

https://github.com/OpenZeppelin/openzeppelin-contracts/commit/d693d89d99325f395182e4f547dbf5ff8e5c3c87

Screen Shot 2022-10-03 at 10 09 24 AM

File: contracts/Baal.sol [hunk 3]
```diff
@@ -268,7 +277,7 @@ contract Baal is Module, EIP712, ReentrancyGuard, BaseRelayRecipient {
                 _initializationMultisendData,
                 Enum.Operation.DelegateCall
             ),
-            "call failure"
+            "call failure setup"
```
Comment (brossetti1) [inline contracts/Baal.sol:281]: adding one more word to better distinguish errors -- there were two `call failure` errors in this file from different functions.

File: contracts/Baal.sol [hunk 4]
```diff
@@ -318,27 +327,25 @@ contract Baal is Module, EIP712, ReentrancyGuard, BaseRelayRecipient {
...
+            expiration,
+            baalGas,
+            0, /* yes votes */
+            0, /* no votes */
+            0, /* highestMaxSharesAndLootAtYesVote */
+            [false, false, false, false], /* [cancelled, processed, passed, actionFailed] */
+            selfSponsor ? _msgSender() : address(0),
+            proposalDataHash,
+            details
+        );
```
Comment (brossetti1) [inline contracts/Baal.sol:351]: qsp for removing the unchecked call as the QuantStamp team pointed out that there are some holes with `unchecked` overflow here so its safer to have it checked

File: contracts/Baal.sol [hunk 5]
```diff
@@ -573,7 +585,7 @@ contract Baal is Module, EIP712, ReentrancyGuard, BaseRelayRecipient {
         bytes calldata _data
     ) external baalOnly {
         (bool success, ) = _to.call{value: _value}(_data);
-        require(success, "call failure");
+        require(success, "call failure execute");
```
Comment (brossetti1) [inline contracts/Baal.sol:596]: other error update

File: contracts/Baal.sol [hunk 6]
```diff
@@ -715,17 +733,19 @@ contract Baal is Module, EIP712, ReentrancyGuard, BaseRelayRecipient {
...
             sharesToken.unpause();
+            emit SharesPaused(false);
         }
+
         if(pauseLoot && !lootToken.paused()){
             lootToken.pause();
+            emit LootPaused(true);
         } else if(!pauseLoot && lootToken.paused()){
             lootToken.unpause();
+            emit LootPaused(false);
```
Comment (brossetti1) [inline contracts/Baal.sol:755]: only emitting events if status changed on `paused`

File: test/BaalSafe.test.ts [hunk 7]
```diff
@@ -2528,42 +2526,63 @@ describe("Baal contract", function () {
...
+        chainId,
+        baal.address,
+        summoner,
+        deploymentConfig.TOKEN_NAME,
+        expiry,
+        1,
+        1,
+        true
+      );
+      const sigTwo = await ethers.utils.splitSignature(signatureTwo);
```
Comment (brossetti1) [inline test/BaalSafe.test.ts:2581]: interesting how this test included a check on nonces inherently as it was using the same signature for both `submitVoteWithSig` - after adding a second signature with the nonce bumped, the test resolved from the error that triggered from adding the nonce. ie notice how nonce had to go from 0 to 1

File: README.md [hunk 8]
```diff
@@ -92,6 +92,38 @@ the dao contracts and the Safe treasury and one to use an existing Safe treasury
...
+    }
+  ]
+}
+```
+
+then run the coverage command:
+
+```
+npx hardhat coverage
+```
```
Comment (brossetti1) [inline README.md:125]: coverage docs - should be turned on manually and turned off for faster test runs

**Review by dekanbro — APPROVED** [review]

**Pull request merged**